common/src/main/java/com/wynntils/gui/screens/CharacterSelectorScreen.java

```diff
@@ -0,0 +1,401 @@
+                0,
+                0,
+                Texture.XP_BAR.width(),
+                Texture.XP_BAR.height(),
+                selected.getClassInfo().xp() / 100f);
+
+        poseStack.pushPose();
+        poseStack.translate(renderWidth * 0.08f, renderHeight * 0.15f, 0);
+
+        RenderUtils.drawScalingTexturedRect(
+                poseStack,
+                Texture.SOUL_POINT_ICON.resource(),
+                0,
+                0,
+                0,
+                Texture.SOUL_POINT_ICON.width() * currentTextureScale,
+                Texture.SOUL_POINT_ICON.height() * currentTextureScale,
+                Texture.SOUL_POINT_ICON.width(),
+                Texture.SOUL_POINT_ICON.height());
+
+        poseStack.pushPose();
+        poseStack.scale(scale, scale, 0f);
+        poseStack.translate(renderWidth * 0.125f / scale, 0, 0);
+        FontRenderer.getInstance()
+                .renderText(
+                        poseStack,
+                        String.valueOf(this.selected.getClassInfo().soulPoints()),
+                        0,
+                        0,
+                        CommonColors.BLACK,
+                        HorizontalAlignment.Left,
+                        VerticalAlignment.Top,
+                        FontRenderer.TextShadow.NONE);
+        poseStack.popPose();
+
+        poseStack.translate(renderWidth * 0.27f, 0, 0);
+        RenderUtils.drawScalingTexturedRect(
+                poseStack,
+                Texture.QUESTS_ICON.resource(),
+                0,
+                0,
+                0,
+                Texture.QUESTS_ICON.width() * currentTextureScale,
+                Texture.QUESTS_ICON.height() * currentTextureScale,
+                Texture.QUESTS_ICON.width(),
+                Texture.QUESTS_ICON.height());
+
+        poseStack.pushPose();
+        poseStack.scale(scale, scale, 0f);
+        poseStack.translate(renderWidth * 0.15f / scale, 0, 0);
+        FontRenderer.getInstance()
+                .renderText(
+                        poseStack,
+                        String.valueOf(this.selected.getClassInfo().completedQuests()),
+                        0,
+                        0,
+                        CommonColors.BLACK,
+                        HorizontalAlignment.Left,
+                        VerticalAlignment.Top,
+                        FontRenderer.TextShadow.NONE);
+        poseStack.popPose();
+
+        poseStack.translate(renderWidth * 0.32f, 0, 0);
+        RenderUtils.drawScalingTexturedRect(
+                poseStack,
+                Texture.CHALLENGES_ICON.resource(),
+                0,
+                0,
+                0,
+                Texture.CHALLENGES_ICON.width() * currentTextureScale,
+                Texture.CHALLENGES_ICON.height() * currentTextureScale,
+                Texture.CHALLENGES_ICON.width(),
+                Texture.CHALLENGES_ICON.height());
+        // FIXME: I have no idea what to render here..
```

Comment:
? Is something still needed, or can you remove this comment?

Reply:
I am not sure what the medal thing stands for.. 
![image](https://user-images.githubusercontent.com/49001742/193426780-cef3f37c-5f5c-41e6-bd82-02634c3f5b3a.png)

I will check legacy.

Reply:
I updated to TODO. I cannot add/test this right now as I do not have free class slots on my main, and my alt is not leveled up.